aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -438,52 +493,59 @@ sub pxelink
 }
 
 # Prints the status of the node.
+# Display information for both PXELINUX and Grub2 variant.
+
 sub Status
 {
     my ($self, $cfg) = @_;
 
     my $t = $cfg->getElement (ETH)->getTree;
-    my $dir = $this_app->option (NBPDIR);
-    my $fqdn = get_fqdn($cfg);
-    my $boot = $this_app->option (LOCALBOOT);
-    my $rescue = link_filepath($cfg, RESCUE);
-    my $firmware = link_filepath($cfg, FIRMWARE);
-    my $livecd = link_filepath($cfg, LIVECD);
-    foreach my $s (values (%$t)) {
-        next unless $s->{ip};
-        my $ln = hexip ($s->{ip});
-        my $since = "unknown";
-        my $st;
-        if (-l "$dir/$ln") {
-            $since = ctime(lstat("$dir/$ln")->ctime());
-            my $name = readlink ("$dir/$ln");
-            my $name_path = "$dir/$name";
-            if (! -e $name_path) {
-                $st = "broken";
-            } elsif ($name =~ m{^(?:.*/)?$fqdn\.cfg$}) {
-                $st = "install";
-            } elsif ($name =~ m{^$boot$}) {
-                $st = "boot";
-            } elsif ($firmware && ($name_path =~ m{$firmware})) {
-                $st = "firmware";
-            } elsif ($livecd && ($name_path =~ m{$livecd})) {
-                $st = "livecd";
-            } elsif ($rescue && ($name_path =~ m{$rescue})) {
-                $st = "rescue";
+
+    foreach my $variant (PXE_VARIANT_PXELINUX, PXE_VARIANT_GRUB2) {
+        my $dir = $this_app->option ($VARIANT_PARAMS[$variant]->{nbpdir_opt});
+        my $boot = $this_app->option (LOCALBOOT);
+        my $fqdn = get_fqdn($cfg);
+        my $rescue = link_filepath($cfg, RESCUE, $variant);
+        my $firmware = link_filepath($cfg, FIRMWARE, $variant);
+        my $livecd = link_filepath($cfg, LIVECD, $variant);
+        foreach my $s (values (%$t)) {
```

Comment:
sort the values, and i know this isn't your code, but single letter variables should be avoided.
`$t` -> $interfaces`; `$s` -> `$interface`